apps/meteor/client/hooks/useJoinRoom.ts

```diff
@@ -0,0 +1,19 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { sdk } from '../../app/utils/client/lib/SDKClient';
+
+export const useJoinRoom = () => {
+	const queryClient = useQueryClient();
+
+	return useMutation({
+		mutationFn: async ({ rid, type }: { rid: IRoom['_id']; type: IRoom['t'] }) => {
+			await sdk.call('joinRoom', rid);
+
+			return { reference: rid, type };
```

Comment:
Are you using this return anywhere? If not we don't need the mutation to return anything IMO.